### ISSUE TITLE
EPUB Book Export

### ISSUE THREAD

**Issue by nosbig** [BookStackApp/BookStack#1949]
Description:
I would like the ability to export an entire book directly to an epub format suitable for delivering to an ebook reader for offline viewing.

Benefits: This would allow for offline viewing of a book on an ebook reader, smart phone, or PC.

**Comment by MikeyMJCO** [comment]
This is a duplicate - find the other open issue on this subject and add your support there. It's not helpful to open duplicate issues.

**Comment by nosbig** [comment]
Yeah, I did search for epub, and I found issue #43, but it seems to be focused more on importing items than an output format.  It's easy for different support channels to bleed together.

**Comment by MikeyMJCO** [comment]
So - I've looked into this - there's one (OLD) PHP class that does epub exports - it doesn't work with PHP 7.3/7.4 as far as I can tell (or any PHP 7.x version!)

MediaWiki have an epub export extension - I'm going to take a look at that and see what they're doing under the hood (if it's still current!)

**Comment by knut02** [comment]
Would it perhaps be possible to do something along the lines of what pressbooks do, use underlaying software installed on the server to export to different formats. [Install instructions](https://docs.pressbooks.org/installation/)  show the different types of software you need to have installed to be able to export to the different formats.

**Comment by ErroneousBosch** [comment]
ePub is an open standard, and the [spec is available here](https://idpf.org/epub/30/spec/epub30-overview.html). Building one from scratch isn't difficult, mostly some templated XML and a folder structure. The container is just a renamed ZIP file. You may already be partway there with your contained HTML export.

There are some libraries on packagist that can make this potentially easier. I might poke at a hack if I have time. 

 More in-depth structure info here: https://www.hxa.name/articles/content/epub-guide_hxa7241_2007.html

**Comment by ssddanbrown** [comment]
BookStack is now [managed on Codeberg](https://codeberg.org/bookstack/bookstack).

As part of this change, we're closing all existing GitHub issues and pull requests.
You'll instead find this issue on Codeberg here: 
https://codeberg.org/bookstack/bookstack/issues/1949
  
Any further discussion for this will now be done on Codeberg.
Therefore this GitHub thread will be locked.
